### PR TITLE
Articulated object semantics

### PR DESCRIPTION
**Pull request by 0mdc**
## Motivation and Context

This changeset adds support for semantic rendering of skinned articulated objects.

The semantic ID of an articulated object can be defined in the corresponding `ao_config.json` file as such:
```
{
    "render_asset": "human.glb",
    "semantic_id": 100
}
```

The identifiers can be specified in a semantic map, e.g. `object_semantic_id_mapping.json` as such:
```
{
    <...>
    "table": 98,
    "plant": 99,
    "human": 100,
    <...>
}
```

human_semantics

human_semantics_rigidbodies

## How Has This Been Tested

Tested locally and in `SimTest::testArticulatedObjectSkinned`.

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have completed my CLA (see **CONTRIBUTING**)
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/esp/io/URDFParser.cpp [hunk 1]
```diff
@@ -193,11 +207,6 @@ bool Model::loadJsonAttributes(const std::string& filename) {
     return (numConfigSettings > 0);
   }  // if has user_defined tag
 
-  if (!hasRenderAsset) {
-    ESP_WARNING() << "<Model> : Json Config file exists but \"" << subGroupName
-                  << "\" and \"" << attrRenderAsset
-                  << "\" tags not found within file.";
-  }
```
Comment (jturner65) [inline src/esp/io/URDFParser.cpp:200]: Is getting rid of this safe?

File: src/esp/io/URDFParser.cpp [hunk 2]
```diff
@@ -149,9 +150,22 @@ bool Model::loadJsonAttributes(const std::string& filename) {
...
       m_renderAsset = std::string(renderAssetPath.data());
-      hasRenderAsset = true;
     }
   }
+
+  // Parse semantic_id
+  if (jsonConfig.HasMember(attrSemanticId)) {
+    if (!jsonConfig[attrSemanticId].IsInt()) {
+      ESP_WARNING() << "<Model> : Json Config file specifies semantic_id "
+                       "attribute but it is not an it. Skipping semantic_id "
```
Comment (jturner65) [inline src/esp/io/URDFParser.cpp:160]: an int?

**Review by jturner65 — APPROVED** [review]
LGTM

**Review by Skylion007 — APPROVED** [review]

File: src/esp/io/URDFParser.cpp (hunk 1 above)
Comment (0mdc) [reply src/esp/io/URDFParser.cpp:200]: I don't think that this warning is necessary. Furthermore, we may want to handle these attributes like we handle the others, therefore moving out this code.

I'll leave this warning out.

**Pull request merged**